Tests/UnitTests/Plugins/Json/AlgebraJsonConverterTests.cpp

```diff
@@ -0,0 +1,90 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Plugins/Json/AlgebraJsonConverter.hpp"
+#include "Acts/Tests/CommonHelpers/DataDirectory.hpp"
+
+#include <fstream>
+#include <iostream>
+
+#include <nlohmann/json.hpp>
+
+BOOST_AUTO_TEST_SUITE(AlgebraJsonConverter)
+
+BOOST_AUTO_TEST_CASE(TransformRoundTripTests) {
+  Acts::Transform3 reference = Acts::Transform3::Identity();
+
+  std::ofstream out;
+
+  // Test the identity transform
+  nlohmann::json identityOut;
+  to_json(identityOut, reference);
+  out.open("Transform3_Identity.json");
+  out << identityOut.dump(2);
+  out.close();
+
+  auto in = std::ifstream("Transform3_Identity.json",
+                          std::ifstream::in | std::ifstream::binary);
+  BOOST_CHECK(in.good());
+  nlohmann::json identityIn;
+  in >> identityIn;
+  in.close();
+
```

Comment:
Do you really have to write the json to a file then read that file again ? The reason I dit it that way for the material was because creating a detector with material for the test was quite cumbersome but here I am not sure it is useful. (same apply for the BU)

Reply:
I think it's nice to check the full round trip, right?

Reply:
Yeah, you are right. I was wondering if the writing of the file could pose a problem during the CI but probably not.